custom_components/switchbot-curtain/cover.py

```diff
@@ -160,7 +157,18 @@ def current_cover_position(self):
         """Return the current position of cover shutter."""
         return self._device.get_position()
 
-    def update(self):
-        self._device.update()
-        self._light = self._device.get_light_level()
-        self._battery = self._device.get_battery_percent()
+    @property
+    def device_info(self):
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self._mac.replace(":", ""))},
+            "name": self._name,
+            "model": "Curtain",
+            "manufacturer": MANUFACTURER,
+        }
+
+#    def update(self):
+#        """Update device attributes."""
+#        self._device.update()
+#        self._light = self._device.get_light_level()
+#        self._battery = self._device.get_battery_percent()
```

Comment:
Found a problem with this code. Could be caused by multiple connection attempts locking the bluetooth adapter.

Working on fix.

Reply:
Fix issue. It now works well with multiple devices. ;-)